### DISCUSSION TITLE
Format on Save in HTML - Closing Tags XHTML style instead of HTML5

### DISCUSSION THREAD

**Discussion by mnmalist** [zed-industries/zed#23944] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

![Shot 2024-09-24 at 00 10 43](https://github.com/user-attachments/assets/94758f71-d368-4b44-a110-066f3030e1c0)

In HTML, when Format on Save is enabled, HTML tags that were closed with `>` are being reformatted to XHTML style` />`.

For example, if the code is written like this:
```
<link rel="stylesheet" href="style.css">
<br>
```

it will always reformat to be closed XHTML style ('/>'), like this:
```
<link rel="stylesheet" href="style.css" />
<br />
```

While the self-closing form is valid, the more conventional and common way in HTML5 is to write it as '>', without the trailing slash.

### Environment

Zed 0.153.6
macOS Sequoia 15.0
Chip: Apple M1 Pro

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by maxdeviant** [comment]
This would need to be something supported by Prettier.

I don't see a configuration option relating to this, and it seems that they're [philosophically against it](https://github.com/prettier/prettier/issues/5641), at least as of 2018.

**Comment by mnmalist** [comment]
![Shot 2024-09-24 at 22 15 06@2x](https://github.com/user-attachments/assets/03b44fc1-ee6a-4a7d-a6fc-0423fd550c7d)

@maxdeviant  Maybe formatting should be handled by "HTML Language Features" then. It has no strong opinion :)

I checked the defaults in VS Code and Fleet, they both have this reformat feature out of the box and it does not do multiple line breaks (for which I saw people complaining about in other issues), nor close inline tags with forward slash.

That could be the right way to do it out of the box, and Prettier should probably be available as a plugin.

**Comment by jgagne** [comment]
Thanks for adding this configuration option.

> [!TIP]
As a workaround, to change HTML void elements from XHTML-style to HTML5-style (removing trailing slashes like `<hr />` → `<hr>`), add something like this to your `settings.json` (without the foo/bar examples):

```json
{
  "example_config_foo": "false",
  "prettier": {
    "bracketSameLine": false,
    "singleQuote": false
  },
  "example_config_bar": "false"
}
```
